integration-tests/src/main/java/com/example/ImplicitModuleInstanceCannotBeCreated.java

```diff
@@ -0,0 +1,26 @@
+package com.example;
+
+import dagger.Component;
+import dagger.Module;
+import dagger.Provides;
+
+@Component(modules = ImplicitModuleInstanceCannotBeCreated.Module1.class)
+public interface ImplicitModuleInstanceCannotBeCreated {
+
+  String string();
+
+  @Module
+  class Module1 {
+
+    Module1() {
+      throw new IllegalStateException("No need to instantiate this");
```

Comment:
Dagger generated code doesn't instantiate this, as it's only calling the static method.
`justAnInstanceMethod` is not called by generated Dagger code, so no need for an instance.